main.py

```diff
@@ -292,6 +293,30 @@ def disable_switch(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         self.notify_switch_links_status(switch, "link disabled")
         return JSONResponse("Operation successful", status_code=201)
+    
+    @rest('v3/switches/{dpid}', methods=['DELETE'])
+    def delete_switch(self, request: Request) -> JSONResponse:
+        """Delete switch"""
+        dpid = request.path_params["dpid"]
+        with self._switch_lock:
+            try:
+                switch: Switch = self.controller.switches[dpid]
+            except KeyError:
+                raise HTTPException(404, detail="Switch not found.")
+            if switch.status != EntityStatus.DISABLED:
+                raise HTTPException(409, detail=f"The switch is not disable.")
+            links = self.links.copy()
```

Comment:
The shallow copy can work here too, but since you'd want an atomic read to avoid other threads potentially changing `self.links` concurrently, you'd need to acquire `self._links_lock` during this for loop, acquiring `self_link_lock` before iterating here would make this endpoint to behave deterministically with fewer potential surprises, although that's already a very rare corner case.

Reply:
Now using `_links_lock`